File: CS/WindowsApplication12/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;

namespace WindowsApplication12
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            ProcessAllEditors(this);
        }

        private void ProcessAllEditors(Control container)
        {
            if (container is BaseEdit)
                //subscribe to the current control's events
                container.GotFocus += new EventHandler(c_GotFocus);
            else
                //process all child controls recursively
                foreach (Control c in container.Controls)
                    ProcessAllEditors(c);
        }

        void c_GotFocus(object sender, EventArgs e)
        {
            if (sender is TextEdit)
                (sender as TextEdit).SelectAll();
        }
    }
}

File: VB/WindowsApplication12/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors

Namespace WindowsApplication12
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			ProcessAllEditors(Me)
		End Sub

		Private Sub ProcessAllEditors(ByVal container As Control)
			If TypeOf container Is BaseEdit Then
				'subscribe to the current control's events
				AddHandler container.GotFocus, AddressOf c_GotFocus
			Else
				'process all child controls recursively
				For Each c As Control In container.Controls
					ProcessAllEditors(c)
				Next c
			End If
		End Sub

		Private Sub c_GotFocus(ByVal sender As Object, ByVal e As EventArgs)
			If TypeOf sender Is TextEdit Then
                CType(sender, TextEdit).SelectAll()
			End If
		End Sub
	End Class
End Namespace